_posts/2018-04-03-EDK2-based-firmware-on-MinnowBoard.md

```diff
@@ -0,0 +1,56 @@
+Building EDK2 based firmware for MinnowBoard
+============================================
+
+There are some options to build firmware for MinnowBoard, a Bay Trail based
+SBC (Single Board Computer) from Intel. We prefer usually coreboot as simplest
+and fastest, open source solution, but sometimes we want to have UEFI
+interface.
+
+UEFI itself doesn't cover whole boot procedure, so its open source reference
+implementation, EDK2 is not enough to build firmware for hardware plafrorm,
+we need to provide PI (Platform Initialization) phase implementation. In EDK2
+repository we can find only implementation for virtualization (OVMF), this
+option is covered in [this article](https://3mdeb.com/firmware/uefi-application-development-in-ovmf/#.WsOfOkuxVuE).
+
+coreboot could be used to provide PI phase, but this procedure is mostly covered
+in [article on building coreboot for MinnowBoard](#Umiescic_link_tutaj), only we
+need to choose Tianocore payload. In this article we cover building UEFI
+firmware using binary objects from Intel. Whole procedure can be done using
+following script:
```

Comment:
coreboot could be used to provide PI phase, but this procedure is mostly covered in the article on building coreboot for MinnowBoard, only we need to choose Tianocore payload. In this article, we cover building UEFI firmware using binary objects from Intel. Whole procedure can be done using following script: